taiga_contrib_ldap_auth_ext/connector.py

```diff
@@ -47,35 +48,94 @@ class LDAPUserLoginError(LDAPError):
+        print(error_message, flush=True)
+        raise LDAPUserLoginError({"error_message": error_message})
+    # we are only interested in user objects in the response
+    print('3', flush=True)
+    connection.response = [r for r in connection.response if 'raw_attributes' in r and 'dn' in r]
+    print('4', flush=True)
+    # stop if no search results
+    if not connection.response:
+        raise LDAPUserLoginError({"error_message": "LDAP login not found"})
+
+    # handle multiple matches
+    if len(connection.response) > 1:
+        raise LDAPUserLoginError(
+            {"error_message": "LDAP login could not be determined."})
+
+    # handle missing mandatory attributes
+    raw_attributes = connection.response[0].get('raw_attributes')
+    if raw_attributes.get(USERNAME_ATTRIBUTE) or raw_attributes.get(EMAIL_ATTRIBUTE) or raw_attributes.get(
+            FULL_NAME_ATTRIBUTE):
+        raise LDAPUserLoginError({"error_message": "LDAP login is invalid."})
+
+    # attempt LDAP bind
+    username = raw_attributes.get(USERNAME_ATTRIBUTE)[0].decode('utf-8')
+    email = raw_attributes.get(EMAIL_ATTRIBUTE)[0].decode('utf-8')
+    full_name = raw_attributes.get(FULL_NAME_ATTRIBUTE)[0].decode('utf-8')
+    try:
+        dn = str(bytes(connection.response[0].get('dn'), 'utf-8'), encoding='utf-8')
+        print(dn, flush=True)
+        print("before connection", flush=True)
+        print(ldap_value, flush=True)
+        Connection(ldap_value, auto_bind=auto_bind, client_strategy=SYNC,
+                   check_names=True, authentication=SIMPLE,
+                   user=dn, password=password)
+        print("after connection",flush=True)
+    except Exception as e:
+        error = "LDAP bind failed: %s" % e
+        raise LDAPUserLoginError({"error_message": error})
+
+    # LDAP binding successful, but some values might have changed, or
+    # this is the user's first login, so return them
+    return username, email, full_name
+
+
 def login(username: str, password: str) -> tuple:
     """
     Connect to LDAP server, perform a search and attempt a bind.
-
     Can raise `exc.LDAPConnectionError` exceptions if the
     connection to LDAP fails.
-
     Can raise `exc.LDAPUserLoginError` exceptions if the
     login to LDAP fails.
-
     :returns: tuple (username, email, full_name)
-
     """
 
     tls = None
     if TLS_CERTS:
         tls = TLS_CERTS
 
-    # connect to the LDAP server
-    if SERVER.lower().startswith("ldaps://"):
-        use_ssl = True
-    else:
-        use_ssl = False
-    try:
-        server = Server(SERVER, port=PORT, get_info=NONE,
-                        use_ssl=use_ssl, tls=tls)
-    except Exception as e:
-        error = "Error connecting to LDAP server: %s" % e
-        raise LDAPConnectionError({"error_message": error})
+    # Dict with server key and connection value
+    server_ldap_list = []
+    # Connect to the LDAP servers
+    servers_list = SERVERS.split(',')
+    for ser in servers_list:
```

Comment:
@AminovE99 Rename `ser` to `server_address`  or `server_host` or `host` or `address`